sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/databuilder/CaseShareDataBuilder.java

```diff
@@ -56,13 +59,22 @@
 	private SampleService sampleService;
 	@EJB
 	private ShareDataBuilderHelper dataBuilderHelper;
+	@Inject
+	private SormasToSormasConfig sormasToSormasConfig;
 
 	public CaseShareData buildShareData(Case caze, User user, SormasToSormasOptionsDto options) throws SormasToSormasException {
 		Pseudonymizer pseudonymizer = dataBuilderHelper.createPseudonymizer(options);
 
 		PersonDto personDto = dataBuilderHelper.getPersonDto(caze.getPerson(), pseudonymizer, options);
 		CaseDataDto cazeDto = getCazeDto(caze, pseudonymizer);
 
+		// external tokens ("Aktenzeichen") are not globally unique in Germany due to SurvNet, therefore, do not
+		// transmit the token to other GAs, but let them generate their own token based on their local, configurable
+		// format
+		if (!sormasToSormasConfig.getRetainExternalToken()){
```

Comment:
Do we need this only for cases?

if yes then the option could be renamed to `retainCaseExternalToken` otherwise  the externalToken should be cleared for `persons` and `contacts` too

Reply:
Persons do not currently have tokens in SurvNet. If need be proper logic needs to be added alter